crates/biome_js_analyze/src/lint/nursery/use_prefer_for.rs

```diff
@@ -0,0 +1,172 @@
+
+                match first_argument.as_any_js_expression()? {
+                    AnyJsExpression::JsArrowFunctionExpression(arg) => {
+                        let prefer_for_or_index = match arg.parameters().ok()? {
+                            AnyJsArrowFunctionParameters::AnyJsBinding(_) => false,
+                            AnyJsArrowFunctionParameters::JsParameters(params) => {
+                                params.items().len() != 1
+                                    || params
+                                        .items()
+                                        .first()
+                                        .and_then(|item| item.ok())
+                                        .is_some_and(|param| param.as_js_rest_parameter().is_some())
+                            }
+                        };
+
+                        if prefer_for_or_index {
+                            return Some(ReportType::PreferForOrIndex(node.clone()));
+                        } else {
+                            // The map fn doesn't take an index param, so it can't possibly be an index-keyed list. Use <For />.
+                            // The returned JSX, if it's coming from React, will have an unnecessary `key` prop to be removed in
+                            // the useless-keys rule.
+                            return Some(ReportType::PreferFor(node.clone()));
+                        }
+                    }
+                    AnyJsExpression::JsFunctionExpression(arg) => {
+                        arg.parameters().ok().map(|params| {
+                            let is_valid = params.items().len() == 1
+                                && params
+                                    .items()
+                                    .first()
+                                    .and_then(|item| item.ok())
+                                    .is_some_and(|param| param.as_js_rest_parameter().is_none());
+
+                            if is_valid {
+                                ReportType::PreferFor(node.clone())
+                            } else {
+                                ReportType::PreferForOrIndex(node.clone())
+                            }
+                        });
+                    }
+                    _ => return None,
+                }
+            }
+
+            return None;
+        }
+
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+
+        let diagnostic = RuleDiagnostic::new(
+            rule_category!(),
+            node.syntax().text_trimmed_range(),
+            markup! {
+            "Array#map causes DOM elements to be recreated."
+            },
+        );
+
+        if matches!(state, ReportType::PreferForOrIndex(_)) {
+            Some(diagnostic.note(markup! {
+                "Use Solid's `<For />` component or `<Index />` component for rendering lists."
```

Comment:
As for the rule pillars, we need to provide three things. This is missing the first one, where we notify the error.

Also, the For and Index could be hyperlinks to the documentation of the components

Reply:
ok, i add here